ledger/src/blockstore_db.rs

```diff
@@ -1613,8 +1611,9 @@ fn get_db_options(access_type: &AccessType) -> Options {
 
 // Returns whether automatic compactions should be disabled based upon access type
 fn should_disable_auto_compactions(access_type: &AccessType) -> bool {
-    // Disable automatic compactions in maintenance mode to prevent accidental cleaning
-    matches!(access_type, AccessType::PrimaryOnlyForMaintenance)
+    // Leave automatic compactions enabled (do not disable) in Primary mode;
+    // disable in all other modes to prevent accidental cleaning
+    !matches!(access_type, AccessType::Primary)
```

Comment:
I like this change!

We might also consider moving the inline comment to the function comment block.